packages/termux-api/0001-Add-termux-usb-library.patch

```diff
@@ -0,0 +1,558 @@
++# find_package(Protobuf REQUIRED CONFIG) for some reason..
++find_package(Protobuf REQUIRED)
++protobuf_generate(
++    LANGUAGE C
++    GENERATE_EXTENSIONS .pb-c.c
++    OUT_VAR PROTO_SRCS
++    PROTOS proto/UsbAPI.proto
++)
++add_library(termux-usb SHARED termux-usb.c termux-usb.h ${PROTO_SRCS} ${PROTO_HDRS})
++
++target_include_directories(
++  termux-usb
++  PUBLIC ${Protobuf_INCLUDE_DIRS} ${libusb_INCLUDE_DIRS}
++  PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/proto ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_BINARY_DIR} ${CMAKE_SOURCE_DIR}
++)
++target_link_libraries(termux-usb PUBLIC ${Protobuf_LIBRARIES} ${CMAKE_BINARY_DIR}/libtermux-api.so -lprotobuf-c ${libusb_LIBRARIES})
++
++install(
++  FILES
++    ${CMAKE_CURRENT_BINARY_DIR}/libtermux-usb.so
++  TYPE LIB
++)
++
++install(
++  FILES ${CMAKE_CURRENT_SOURCE_DIR}/termux-usb.h
++  TYPE INCLUDE
++)
+diff --git a/usb/proto/UsbAPI.proto b/usb/proto/UsbAPI.proto
+new file mode 100644
+index 000000000000..5a1d89c5cb6a
+--- /dev/null
++++ b/usb/proto/UsbAPI.proto
+@@ -0,0 +1,57 @@
++syntax = "proto3";
++
++package usbapi;
++
++option java_package = "com.termux.api";
++option java_outer_classname = "UsbAPIProto";
++
++/* Modelled after libusb's struct libusb_endpoint_descriptor */
++message termuxUsbEndpointDescriptor {
++  int32 endpointAddress = 1;
++  int32 attributes      = 2;
++  int32 maxPacketSize   = 3;
++  int32 interval        = 4;
++}
++
++/* Modelled after libusb's struct libusb_interface_descriptor */
++message termuxUsbInterfaceDescriptor {
++  int32  alternateSetting                       = 1;
++  int32  interfaceClass                         = 2;
++  int32  interfaceSubclass                      = 3;
++  int32  interfaceProtocol                      = 4;
++  string interface                              = 5;
++  repeated termuxUsbEndpointDescriptor endpoint = 6;
++}
++
++/* Modelled after libusb's struct libusb_config_descriptor */
++message termuxUsbConfigDescriptor {
++  int32  configurationValue                       = 1;
++  int32  maxPower                                 = 2;
++  string configuration                            = 3;
++  repeated termuxUsbInterfaceDescriptor interface = 4;
++}
++
++/* Modelled after libusb's struct libusb_device_descriptor */
++message termuxUsbDeviceDescriptor {
++  int32  configurationCount = 1;
++  int32  deviceClass        = 2;
++  int32  deviceProtocol     = 3;
++  int32  deviceSubclass     = 4;
++  int32  productId          = 5;
++  int32  vendorId           = 6;
++  string manufacturerName   = 7;
++  string productName        = 8;
++  string serialNumber       = 9;
++}
++
++/* Loosely modelled after libusb's internal struct libusb_device */
++message termuxUsbDevice {
++  int32  busNumber                           = 1;
++  int32  portNumber                          = 2;
++  string deviceAddress                       = 3;
++  termuxUsbDeviceDescriptor device           = 4;
++}
++
++message termuxUsb {
++  repeated termuxUsbDevice device = 1;
++}
```

Comment:
It is easy to loose this with package update. Probably this file should be somehow downloaded from correct termux/termux-api git branch or at least be versioned and the version should be checked somewhere in the code.

Reply:
Yes, maybe we should have some sort of termux-shared repo with the protobuf definitions. I think Apollo had some ideas already for the upcoming termux-api re-work. Having a version is a good idea, I'll think a bit on where/how to add it!